File: rps/solutions/RpsPattern.java
import java.io.*;

public class RpsPattern {
    static int roundNum = 0;
    
    static char playMove() {
        // This is a different pattern from what was described in the solution!
        // Can you figure out why it also works?
        switch(roundNum % 3) {
            case 0:
                return 'R';
            case 1:
                return 'P';
            default:
                return 'S';
        }
    }
    
    static void readBotMove(char botMove) {
        // This strategy doesn't care about opponent moves, so we do nothing here
    }
    
    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    static PrintWriter out = new PrintWriter(System.out);

    public static void main(String[] args) throws IOException {
        int T = Integer.parseInt(in.readLine());
        for (int i = 0; i < T; i++) {
            out.println(playMove());
            out.flush();
            readBotMove(in.readLine().charAt(0));
            
            roundNum++;
        }
    }
}

File: rps/solutions/rps_pattern.cpp
#include <iostream>

using namespace std;

int round_num = 0;

char play_move() {
    switch(round_num % 6) {
        case 0:
        case 1:
            return 'R';
        case 2:
        case 3:
            return 'S';
        default:
            return 'P';
    }
}

void read_bot_move(char bot_move) {
    // This strategy doesn't care about opponent moves, so we do nothing here
}

int main() {
    int T;
    cin >> T;
    for (int i = 0; i < T; i++) {
        char my_move;
        my_move = play_move();
        cout << my_move << endl;
        
        char bot_move;
        cin >> bot_move;
        read_bot_move(bot_move);
        
        round_num++;
    }
}
